sdk/keyvault/keyvault-certificates/tests/utils/recorder.ts

```diff
@@ -0,0 +1,434 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import fs from "fs-extra";
+import nise from "nise";
+import { retry as realRetry } from "./retry";
+import { isNode as coreIsNode, delay as coreDelay } from "@azure/core-http";
+import queryString from "query-string";
+import * as dotenv from "dotenv";
+dotenv.config({ path: "../.env" });
+
+export function isBrowser(): boolean {
+  return typeof window !== "undefined";
+}
+
+export const isNode = coreIsNode;
+
+export function escapeRegExp(str: string): string {
+  return encodeURIComponent(str).replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+export async function blobToString(blob: Blob): Promise<string> {
+  const fileReader = new FileReader();
+  return new Promise<string>((resolve, reject) => {
+    fileReader.onloadend = (ev: any) => {
+      resolve(ev.target!.result);
+    };
+    fileReader.onerror = reject;
+    fileReader.readAsText(blob);
+  });
+}
+
+let nock: any;
+if (!isBrowser()) {
+  nock = require("nock");
+}
+
+export const env = isBrowser() ? (window as any).__env__ : process.env;
+export const isRecording = env.TEST_MODE === "record";
+export const isPlayingBack = env.TEST_MODE === "playback";
+
+// IMPORTANT: These are my attempts to make this more generic without changing it significantly
```

Comment:
You probably want this comment not in the code but in the PR. Talk about what you changed and why you think the new version is better.

Reply:
This is a copy of the recorder.ts we have in keys and secrets. It will be replaced with a core recorder that Harsha is going to introduce soon. I can remove this comment if that's better!